assets/js/header.js

```diff
@@ -5,4 +5,12 @@ const observer = new IntersectionObserver(
   { threshold: [1] }
 );
 
-observer.observe( headerElement );
\ No newline at end of file
+if ( headerElement ) {
+  observer.observe( headerElement );
+}
+
+const isFullWidthLearningModeTemplate = document.querySelector( '.sensei-course-theme__header .sensei-course-theme-course-progress-bar' );
```

Comment:
I was thinking the same, because found it really hard to differentiate among templates, as you said, it was tricky. To avoid editing templates, thought about inserting the name of the template as a class manipulating string using code where the template files are loaded. But then decided to use JS to not edit all templates in the core plugin and pro now as it was working for now.

But I think we should add those classes later and also remove some inline styles from the templates so that we don't have to struggle overriding those from themes and have to resort to using !important